File: scripts/m42_w46_capture_timeline_ux.mjs
import { chromium } from "playwright";
import path from "path";
import fs from "fs";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const art = path.join(__dirname, "..", "artifacts", "m42", "w46", "timeline-ux");
fs.mkdirSync(art, { recursive: true });

const shots = [
  ["beta-1280-night.png", 1280, 720, "aurora-night"],
  ["beta-1920-night.png", 1920, 1080, "aurora-night"],
  ["beta-1280-day.png", 1280, 720, "aurora-day"],
  ["beta-1920-day.png", 1920, 1080, "aurora-day"],
];

const browser = await chromium.launch({ headless: true });
for (const [name, w, h, theme] of shots) {
  const context = await browser.newContext({ viewport: { width: w, height: h } });
  const page = await context.newPage();
  const projectId = process.env.ADEPT_CAPTURE_PROJECT_ID || "e32dae30-a014-4ea4-a2f2-69f4b7809bde";
  await page.goto(`http://127.0.0.1:8760/project/${projectId}?workspace=timeline`, {
    waitUntil: "domcontentloaded",
    timeout: 30000,
  });
  await page.evaluate((t) => {
    document.documentElement.setAttribute("data-theme", t);
    try {
      localStorage.setItem("adept_ui_theme", t);
    } catch {
      /* ignore */
    }
  }, theme);
  try {
    await page.getByTestId("timeline-editor-shell").waitFor({ timeout: 20000 });
    await page.getByTestId("timeline-track-board").waitFor({ timeout: 15000 });
    // Wait until loader clears (tracks rendered or empty instructional lanes)
    await page.waitForFunction(() => {
      const board = document.querySelector('[data-testid="timeline-track-board"]');
      if (!board) return false;
      const text = board.textContent || "";
      return !text.includes("Loading Timeline tracks");
    }, { timeout: 20000 });
  } catch (e) {
    console.log("nav soft fail", name, String(e.message || e));
    try {
      await page.getByRole("button", { name: /unlock|continue|open/i }).first().click({ timeout: 3000 });
      await page.getByTestId("timeline-editor-shell").waitFor({ timeout: 15000 });
    } catch {
      /* ignore */
    }
  }
  const out = path.join(art, name);
  await page.screenshot({ path: out });
  console.log("captured", name, fs.statSync(out).size);
  await context.close();
}
await browser.close();

File: scripts/m42_w46_capture_timeline_ux.py
#!/usr/bin/env python3
"""Best-effort Beta screenshots for Timeline UX mockup parity."""

from __future__ import annotations

import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
ART = ROOT / "artifacts" / "m42" / "w46" / "timeline-ux"
BASE = "http://127.0.0.1:8760"


def main() -> int:
    try:
        from playwright.sync_api import sync_playwright
    except Exception as exc:
        print(f"playwright unavailable: {exc}")
        return 0

    ART.mkdir(parents=True, exist_ok=True)
    shots = [
        ("beta-1280-night.png", 1280, 720, "aurora-night"),
        ("beta-1920-night.png", 1920, 1080, "aurora-night"),
        ("beta-1280-day.png", 1280, 720, "aurora-day"),
        ("beta-1920-day.png", 1920, 1080, "aurora-day"),
    ]
    try:
        with sync_playwright() as p:
            browser = p.chromium.launch(headless=True)
            for name, w, h, theme in shots:
                context = browser.new_context(viewport={"width": w, "height": h})
                page = context.new_page()
                page.goto(BASE, wait_until="domcontentloaded", timeout=20000)
                page.evaluate(
                    """(theme) => {
                      document.documentElement.setAttribute('data-theme', theme);
                      try { localStorage.setItem('adept_ui_theme', theme); } catch (e) {}
                    }""",
                    theme,
                )
                # Best-effort open Timeline
                try:
                    page.locator("a[href*='/project/'], [data-testid='project-card']").first.click(timeout=5000)
                    page.get_by_role("button", name="Timeline").first.click(timeout=8000)
                    page.get_by_test_id("timeline-editor-shell").wait_for(timeout=15000)
                except Exception:
                    pass
                page.screenshot(path=str(ART / name), full_page=False)
                context.close()
                print(f"captured {name}")
            browser.close()
    except Exception as exc:
        print(f"capture failed: {exc}")
        return 0
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
